be/src/vec/functions/like.h

```diff
@@ -166,11 +169,64 @@ class FunctionLikeBase : public IFunction {
                                                  const StringValue* values, uint16_t size,
                                                  unsigned char* result);
 
-    static Status constant_regex_fn(LikeSearchState* state, const StringValue& val,
-                                    const StringValue& pattern, unsigned char* result);
+    static Status constant_regex_fn(LikeSearchState* state, const ColumnString& val,
+                                    const StringValue& pattern, ColumnUInt8::Container& result);
+
+    static Status regexp_fn(LikeSearchState* state, const ColumnString& val,
+                            const StringValue& pattern, ColumnUInt8::Container& result);
+
```

Comment:
Add a comment the code with name `fn_predicate` only execute in storage engine？ or a better name ?